### ISSUE TITLE
MDT affix importing should be based per season, not one value fits all

### ISSUE THREAD

**Issue by Wotuu** [Wotuu/keystone.guru#1294]
I have to constantly adjust Conversion.php - it shouldn't be this way. Each affix for a season should have an MDT index assigned to it instead.

**Comment by Wotuu** [comment]
## Implementation plan — data-driven `mdt_index`, remove hardcoded branches

**Goal.** Stop hand-editing `Conversion.php` per season. Give each affix group an explicit `mdt_index` (its MDT week number) and make MDT week↔affix-group conversion a pure data lookup, removing the `Season::SEASON_TWW_S1` special-cases at `app/Logic/MDT/Conversion.php:391-395` and `413-418`.

**1. Schema.**
- New migration `database/migrations/2026_..._add_mdt_index_to_affix_groups_table.php` (follow the pattern in `2026_07_02_000000_add_default_to_aura_in_spells_table.php`): `$table->smallInteger('mdt_index')->nullable()->after('seasonal_index');` with a matching `down()`.

**2. Model.** `app/Models/AffixGroup/AffixGroup.php`: add `mdt_index` to `$fillable` (and the `@property` docblock). It's seeded via the `SeederModel` insert, so ensure the column is part of the inserted attributes.

**3. Seeder — the bulk of the work.** `database/seeders/AffixSeeder.php` builds every affix group in `$groups` (lines ~120-310) and inserts with attributes at ~320-325. Populate `mdt_index` for **every** affix group across all seasons.
- **Source of truth = MDT's own affix-week rotation** (the MDT composer package's Lua data; see the `update-mdt-package` skill), NOT hand-guessed numbers.
- **Recommended (robust) approach:** derive `mdt_index` programmatically by matching each keystone affix group's non-seasonal affixes to MDT's affix-week list, rather than typing a literal per group. `AffixGroup::findMatchingAffixGroupsForAffixIds()` (`app/Models/AffixGroup/AffixGroup.php:70-90`) already does affix-set matching and is a model for the inverse. This avoids trading a hardcoded branch for hardcoded magic numbers.

**4. Conversion.php.** Rewrite both methods to pure lookups, deleting the season branches:
- `convertWeekToAffixGroup(...)`: within the resolved season, `return $season->affixGroups->firstWhere('mdt_index', $mdtWeek);` (keep the season-resolution logic — note #1271 changes that part).
- `convertAffixGroupToWeek(AffixGroup $affixGroup): int`: `return $affixGroup->mdt_index;` (guard null → default MDT week 1 to match the export fallback at `MDTExportStringService.php:435-436`).

**5. `SeasonsSeeder` cleanup (optional, verify first).** `start_affix_group_index` / `affix_group_count` still feed `SeasonAffixGroupService` (affix-overview rotation) — do **not** remove them; only the MDT conversion stops using them.

**Tests.**
- Unit round-trip on `Conversion`: for representative seasons, `convertWeekToAffixGroup(week)` then `convertAffixGroupToWeek(group)` returns the same week.
- Data-integrity test: within each season, `mdt_index` values are unique and cover the expected week range (assert every affix group has a non-null unique `mdt_index` per season).
- Existing MDT tests in `tests/Feature/App/Service/MDT/` currently assert nothing on affix groups — add the affix assertion there too.

**Verify.**
- `docker compose exec -T app php artisan migrate` (worktree shares main DB — non-destructive add only; never `migrate:fresh`).
- Re-seed affixes and run `docker compose exec -T app php artisan test --compact --filter=Conversion`.
- `composer run fix` + `composer run analyse`.

---
<sub>🤖 Investigation + plan generated with [Claude Code](https://claude.com/claude-code)</sub>

**Comment by Wotuu** [comment]
## Handover — point 3 (deriving `mdt_index`) is blocked on missing production data

Picking up the [implementation plan above](#issuecomment-4892384935). Steps 1, 2, 4, 5 (migration, model, `Conversion.php` rewrite, `SeasonsSeeder` leave-alone) are well understood and ready. **Step 3 — populating `mdt_index` per affix group — is blocked.** Recording findings so this can resume cleanly.

### Approach that was attempted (per the production-dump idea)
The `docker-compose/data/keystoneguru_production.sql` dump contains **only the `mdt_imports` table** (3381 inserts, ids up to 267513): `id, dungeon_route_id, error, import_string, created_at, updated_at`. Date range of imports: **2019-02-03 → 2026-07-06**.

Decoding a sample `import_string` via `MDTImportStringService::setEncodedString()->getDecoded()` yields top-level keys `objects, week, value, text`:
- `week` = the MDT affix week (1-based) ✅
- `value.currentDungeonIdx` = the dungeon ✅
- **no affix list** — MDT derives affixes from `week` via its own rotation, so the string alone cannot tell us which affix group a week corresponds to.

The intended cross-reference is therefore: `import_string → week W` + `dungeon_route → affix_group_id G` ⇒ within `G.season_id`, `week W maps to G` ⇒ `mdt_index(G) = W` (majority-voted over all imports, `error IS NULL`).

### The blocker
This step needs the **route → affix_group link** (`dungeon_route_affix_groups`), but that data is not available:
- The dump excludes `dungeon_routes` / `dungeon_route_affix_groups` (excluded for size).
- The dev DB does **not** contain the production routes. It has 123 seeded sample routes (ids 37–3620); the dump references `dungeon_route_id` up to ~1,133,746. Route 1142 (first row) does not exist locally. So the join resolves for almost nothing here.

### What's needed to resume
1. **The route→affix_group link data.** Only two columns are actually required: `dungeon_route_affix_groups(dungeon_route_id, affix_group_id)` for the referenced routes. A slim dump of that table (or running the derivation on a machine that has the production DB copy) unblocks it. Note: `affix_group` rows are season-specific, so a single `(W, G)` pair already yields `(G.season_id, W→G)` — **`route.created_at` is not needed for the join** (only useful as a hygiene filter to drop imports whose timestamp falls outside `G`'s season).
2. **Fallback for seasons outside the dump's date range.** Empirical derivation can only cover seasons with imports between 2019-02 and 2026-07. Seasons with no/low data (e.g. Legion Timewalking, placeholder start `2050-12-28`; possibly Midnight S1) need an explicit fallback — natural rotation order, or leave `mdt_index` null and let `Conversion` default to week 1.

### Derivation recipe (for whoever runs it)
Throwaway one-time command (not committed; the committed artifact is the values baked into `AffixSeeder`):
- filter `error IS NULL`, decode each `import_string` → `week`
- join to route's single `affix_group_id`
- majority-vote `week` per `(season_id, affix_group_id)` → `mdt_index`
- assert `mdt_index` is unique within each season
- **cross-check derived values against current `Conversion::convertAffixGroupToWeek()`** — the diffs are exactly the seasons the old formula got wrong (the justification for this issue) and make good test evidence.

Reference points (verified this session): hardcoded branches at `app/Logic/MDT/Conversion.php:391-395` (`convertWeekToAffixGroup`) and `413-419` (`convertAffixGroupToWeek`), both keyed on `Season::SEASON_TWW_S1`. Forward path reads `$decoded['week']` at `MDTImportStringService.php:69`; reverse path writes `week` at `MDTExportStringService.php:435-436`. Column would be added to `AffixGroup` `$fillable`, a migration, and both `database/schema/*.dump` files; seeded in the `AffixSeeder` insert loop (~lines 316-343).

---
<sub>🤖 Handover generated with [Claude Code](https://claude.com/claude-code)</sub>
